MapboxDirections/MBApproachType.swift

```diff
@@ -0,0 +1,39 @@
+import Foundation
+
+/**
+ Indicate how a route considers from which side of the road to approach a waypoint.
+ */
+@objc(MBApproachType)
+public enum ApproachType: UInt, CustomStringConvertible {
+    /**
+     Approaches a waypoint from either side of road.
+     */
+    case unrestricted
+
+    /**
+     Approaches a waypoint from the same side of the road as the waypoint falls.
+     */
+    case curb
```

Comment:
A road has two curbs. This option forces the route to approach the curb closest to the waypoint. So rename this case `closerCurb` or `avoidsRoadCrossing`.